From 37a5ba98e5c02472ac66687878b1e669f3d23658 Mon Sep 17 00:00:00 2001
From: Yihao Han <hanyihao@vivo.com>
Date: Tue, 8 Mar 2022 10:41:32 +0100
Subject: [PATCH 352/595] [Backport] media: vidtv: use vfree() for memory
 allocated with vzalloc()

mainline inclusion
from mainline-v5.17-rc1
commit 298cf3dfacc971350be102ddb55bbc2916e32e34
category: bugfix
bugzilla: 187931

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=298cf3dfacc971350be102ddb55bbc2916e32e34

--------------------------------

It is allocated with vzalloc(), the corresponding release function
should not be kfree(), use vfree() instead.

Generated by: scripts/coccinelle/api/kfree_mismatch.cocci

Signed-off-by: Yihao Han <hanyihao@vivo.com>
Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>
Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>
Signed-off-by: Guo Xuenan <guoxuenan@huawei.com>
---
 drivers/media/test-drivers/vidtv/vidtv_s302m.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/test-drivers/vidtv/vidtv_s302m.c b/drivers/media/test-drivers/vidtv/vidtv_s302m.c
index 4676083cee3b..9da18eac04b5 100644
--- a/drivers/media/test-drivers/vidtv/vidtv_s302m.c
+++ b/drivers/media/test-drivers/vidtv/vidtv_s302m.c
@@ -501,7 +501,7 @@ struct vidtv_encoder
 	return e;
 
 out_kfree_buf:
-	kfree(e->encoder_buf);
+	vfree(e->encoder_buf);
 
 out_kfree_e:
 	kfree(e->name);
-- 
2.17.1

